packages/semaphore-signature-kudos-pcd/src/SemaphoreSignatureKudosPCD.ts

```diff
@@ -0,0 +1,279 @@
+ * initialize this PCD package.
+ */
+export interface SemaphoreSignatureKudosPCDInitArgs {
+  /**
+   * This function lets the PCD Card Body UI create a QR code from a PCD, which,
+   * when scanned, directs a scanner to a webpage that verifies whether this PCD
+   * is valid.
+   */
+  makeEncodedVerifyLink?: (encodedPCD: string) => string;
+}
+
+// Stores the initialization arguments for this PCD Package, which are used by
+// either `prove` or `verify` to initialize the required objects the first time either is called.
+export let initArgs: SemaphoreSignatureKudosPCDInitArgs;
+
+/**
+ * Initializes this {@link EdDSAFrogPCDPackage}.
+ */
+async function init(args: SemaphoreSignatureKudosPCDInitArgs): Promise<void> {
+  initArgs = args;
+}
+
+/**
+ * Defines the essential parameters required for creating an {@link EdDSAFrogPCD}.
+ */
+export type SemaphoreSignatureKudosPCDArgs = {
+  identity: PCDArgument<SemaphoreIdentityPCD>;
+  data: ObjectArgument<IKudosData>;
+};
+
+/**
+ * Defines the EdDSA Frog PCD claim. The claim contains data that was signed
+ * with the private key corresponding to the given public key stored in the proof.
+ */
+export interface SemaphoreSignatureKudosPCDClaim {
+  data: IKudosData;
+}
+
+/**
+ * Defines the Semaphore Signature Kudos PCD proof. The proof is an Semaphore Signature PCD whose message
+ * is the encoded data.
+ */
+export interface SemaphoreSignatureKudosPCDProof {
+  semaphoreSignaturePCD: SemaphoreSignaturePCD;
+}
+
+/**
+ * The Semaphore Signature Kudos PCD enables the verification that a specific  {@link SemaphoreSignatureKudosPCDClaim}
+ * has been proven with a Semaphore identity. The {@link SemaphoreSignatureKudosPCDProof} contains a Semaphore Signature
+ * PCD and serves as the proof for this PCD.
+ */
+export class SemaphoreSignatureKudosPCD
+  implements
+    PCD<SemaphoreSignatureKudosPCDClaim, SemaphoreSignatureKudosPCDProof>
+{
+  type = SemaphoreSignatureKudosPCDTypeName;
+  claim: SemaphoreSignatureKudosPCDClaim;
+  proof: SemaphoreSignatureKudosPCDProof;
+  id: string;
+
+  public constructor(
+    id: string,
+    claim: SemaphoreSignatureKudosPCDClaim,
+    proof: SemaphoreSignatureKudosPCDProof
+  ) {
+    this.id = id;
+    this.claim = claim;
+    this.proof = proof;
+  }
+}
+
+/**
+ * Creates a new {@link EdDSAFrogPCD} by generating an {@link EdDSAFrogPCDProof}
+ * and deriving an {@link EdDSAFrogPCDClaim} from the given {@link EdDSAFrogPCDArgs}.
+ */
+export async function prove(
+  args: SemaphoreSignatureKudosPCDArgs
+): Promise<SemaphoreSignatureKudosPCD> {
+  if (!initArgs) {
+    throw new Error("package not initialized");
+  }
+
+  if (!args.identity.value?.pcd) {
+    throw new Error("cannot make kudos proof: identity is not set");
+  }
+
+  if (!args.data.value) {
+    throw new Error("cannot make kudos proof: missing data value");
+  }
+
+  // Custom parser?
+  const seralizedData = JSON.stringify(args.data.value);
```

Comment:
note that JSON.stringify is [not deterministic](https://stackoverflow.com/questions/42491226/is-json-stringify-deterministic-in-v8#:~:text=seen%20(yet%3F)-,JSON.,deterministic%20on%20the%20specification%20level.). consider using something like [json-stable-stringify](https://www.npmjs.com/package/json-stable-stringify)